### PR TITLE
Trim Helm name override for Producer CronJob to 52 characters

### LINKED ISSUE #549: Fix producer-app CronJob name exceeding Kubernetes character limit
Kubernetes limits `metadata.name` for CronJob to 52 characters.

https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#:~:text=The%20name%20of%20a%20CronJob,no%20longer%20than%2052%20characters.

we should implement existing trimming + hashing logic for producer-app to further restrict its `nameOverride`.

### PR DESCRIPTION
**Pull request by disrupted**
Fixes #549

### CODE REVIEW

File: tests/components/streams_bootstrap/test_producer_app.py [hunk 1]
```diff
@@ -60,11 +60,49 @@ def producer_app(self) -> ProducerApp:
                 },
             },
         )
+        assert producer.is_cron_job is False
+        return producer
+
+    @pytest.fixture()
+    def producer_app_cron_job(self) -> ProducerApp:
+        producer = ProducerApp(
```
Comment (raminqaf) [inline tests/components/streams_bootstrap/test_producer_app.py:68]: Should we use model_validate? :) 
```suggestion
        producer = ProducerApp.model_validate(
```

File: tests/components/streams_bootstrap/test_producer_app.py [hunk 2]
```diff
@@ -60,11 +60,49 @@ def producer_app(self) -> ProducerApp:
...
     @pytest.fixture(autouse=True)
     def empty_helm_get_values(self, mocker: MockerFixture) -> MagicMock:
         return mocker.patch.object(Helm, "get_values", return_value=None)
 
+    def test_helm_name_override(self, producer_app: ProducerApp):
+        assert len(producer_app.helm_name_override) == 63
+        assert producer_app.helm_name_override == PRODUCER_APP_HELM_NAME_OVERRIDE
+
+    def test_cron_job_helm_name_override(self, producer_app_cron_job: ProducerApp):
+        assert len(producer_app_cron_job.helm_name_override) == 52
```
Comment (raminqaf) [inline tests/components/streams_bootstrap/test_producer_app.py:100]: I think we have a const for this. How about we use that?

Comment (disrupted) [reply tests/components/streams_bootstrap/test_producer_app.py:100]: not for this new one. The Helm release name is limited to 53 chars, but Kubernetes CronJob is limited to 52

Comment (disrupted) [reply tests/components/streams_bootstrap/test_producer_app.py:100]: also where possible test assertions should be independent from the code

File: tests/components/streams_bootstrap/test_producer_app.py (hunk 1 above)
Comment (disrupted) [reply tests/components/streams_bootstrap/test_producer_app.py:68]: definitely!

File: kpops/components/streams_bootstrap/producer/producer_app.py [hunk 3]
```diff
@@ -70,6 +75,13 @@ def apply_to_outputs(self, name: str, topic: TopicConfig) -> None:
             case _:
                 super().apply_to_outputs(name, topic)
 
+    @property
+    @override
+    def helm_name_override(self) -> str:
+        if self.is_cron_job:
+            return trim(52, self.full_name, "")
```
Comment (raminqaf) [inline kpops/components/streams_bootstrap/producer/producer_app.py:82]: Then maybe set 52 as a const, with the link to the Kubernetes docs?

**Pull request merged**